modules/layers/src/text-layer/text-layer.js

```diff
@@ -113,6 +99,21 @@ export default class TextLayer extends CompositeLayer {
     });
   }
 
+  fontChanged(oldProps, props) {
+    if (
+      oldProps.fontFamily !== props.fontFamily ||
+      oldProps.characterSet !== props.characterSet ||
+      oldProps.fontWeight !== props.fontWeight
+    ) {
+      return true;
+    }
+
+    const oldFontSettings = oldProps.fontSettings || {};
+    const fontSettings = props.fontSettings || {};
+
+    return !FONT_SETTINGS_PROPS.every(prop => oldFontSettings[prop] === fontSettings[prop]);
```

Comment:
- Compare the two objects shallowly first. They do not need to be deep-compared if they are strictly equal.
- `some` is more efficient than `every`.

```js
return oldFontSettings !== fontSettings &&
	FONT_SETTINGS_PROPS.some(prop => oldFontSettings[prop] !== fontSettings[prop]);
```